mmv1/products/compute/Interconnect.yaml

```diff
@@ -0,0 +1,385 @@
+    wait_ms: 10000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'targetLink'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'status'
+    complete: 'DONE'
+    allowed:
+      - 'PENDING'
+      - 'RUNNING'
+      - 'DONE'
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error/errors'
+    message: 'message'
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'compute_interconnect_basic'
+    skip_test: true
+    primary_resource_id: 'example-interconnect'
+    vars:
+      interconnect_name: 'example-interconnect'
+      customer_name: 'example_customer'
+      deletion_protection: 'true'
+    test_vars_overrides:
+      deletion_protection: 'false'
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'compute_interconnect_basic_test'
+    skip_docs: true
+    primary_resource_id: 'example-interconnect'
+    vars:
+      interconnect_name: 'example-interconnect'
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      An optional description of this resource. Provide this property when you create the resource.
+  - !ruby/object:Api::Type::Time
+    name: 'creationTimestamp'
+    description: |
+      Creation timestamp in RFC3339 text format.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      Name of the resource. Provided by the client when the resource is created. The name must be
+      1-63 characters long, and comply with RFC1035. Specifically, the name must be 1-63 characters
+      long and match the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?` which means the first
+      character must be a lowercase letter, and all following characters must be a dash,
+      lowercase letter, or digit, except the last character, which cannot be a dash.
+    required: true
+    immutable: true
+    validation: !ruby/object:Provider::Terraform::Validation
+      regex: '^[a-z]([-a-z0-9]*[a-z0-9])?$'
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    description: |
+      URL of the InterconnectLocation object that represents where this connection is to be provisioned.
+    required: true
+    immutable: true
+  - !ruby/object:Api::Type::Enum
+    name: 'linkType'
+    description: |
+      Type of link requested. Note that this field indicates the speed of each of the links in the
+      bundle, not the speed of the entire bundle. Can take one of the following values:
+        - LINK_TYPE_ETHERNET_10G_LR: A 10G Ethernet with LR optics.
+        - LINK_TYPE_ETHERNET_100G_LR: A 100G Ethernet with LR optics.
+    values:
+      - :LINK_TYPE_ETHERNET_10G_LR
+      - :LINK_TYPE_ETHERNET_100G_LR
+    required: true
+    immutable: true
+  - !ruby/object:Api::Type::Integer
+    name: 'requestedLinkCount'
+    description: |
+      Target number of physical links in the link bundle, as requested by the customer.
+    required: true
+    immutable: true
+  - !ruby/object:Api::Type::Enum
```

Comment:
This could be a string so we don't document the IT_PRIVATE value

Reply:
The official api documentation uses an enum, so this will be okay as is. Thank you for raising the alternative solution.